### ISSUE TITLE
deck.gl and ArcGIS' `SceneView` 3D camera are not aligned properly

### ISSUE THREAD

**Issue by damix911** [visgl/deck.gl#5271]
#### Description
deck.gl and ArcGIS' `SceneView` 3D camera are not aligned properly.

#### Repro Steps
- Open this pen: https://codepen.io/dawken/pen/abmyLoP
- The first `SceneView` uses an Esri VTL basemap
- The second `SceneView` uses the raster version of the same basemap
- Notice how the scale of the deck.gl `ScatterplotLayer` is different compared to the basemap
- Move the camera; both views suffer from alignment issues. Notably, the view with the VTL basemap is good if you don't tilt; the view with the raster basemap shows severe lack of alignment even without tilt.

#### Environment (please complete the following information):
Reported in deck.gl 8.3.0 by @llwinner.

#### Logs
Nothing relevant.

**Comment by wilson0028** [comment]
Is there any timeline as to when this bug may be resolved or is support for ESRI going away?  At NOAA, we are visualizing weather data with deck.gl and unfortunately our mapping contracts are tied to ESRI.

**Comment by damix911** [comment]
You are right @wilson0028 there is still an issue with the zoom level; @Pessimistress made a lot of improvements to the original code and I believe that a lot of cases have been fixed. But there is still a difference between vector tiles basemaps and raster basemaps.

![Screenshot from 2022-04-24 15-00-57](https://user-images.githubusercontent.com/12469086/164998561-9cbea110-5fdd-41e2-aa18-b84a95b15bc4.png)

It could be solved by properly translating the zoom level; we are basically off by a factor of 2 (i.e. a single zoom level).

![Screenshot from 2022-04-24 15-01-22](https://user-images.githubusercontent.com/12469086/164998634-feb5ce31-ade1-443f-bb60-fede89d13d52.png)

**Can you please tell us more about the issue that you are experiencing? Is it happening in local 3D mode with a raster tile basemap?**

Thanks!

**Comment by wilson0028** [comment]
@damix911 Thank you for looking into this issue!  We are using SceneView with vector and raster tiles.  Some vector tiles work like the example below (dark-gray-vector) while others fail (topo-vector)
![works](https://user-images.githubusercontent.com/36017514/165140778-bc4da263-e8d3-4eeb-83c3-16b4d61fae47.png)
Here is an example of a vector layer that doesn't work (topo-vector - notice how my state lines layer doesn't match the ESRI state lines)
![wrong1](https://user-images.githubusercontent.com/36017514/165140980-d9b6b3a0-01a1-4741-929e-611ece3d7faf.png)
Here is an example using a raster ESRI layer (satellite - again notice how the state lines don't match up)
![wrong2](https://user-images.githubusercontent.com/36017514/165141122-b246072e-cb11-45e7-b845-541596188a2d.png)

**Comment by damix911** [comment]
Yes, makes sense, thank you. I am surprised by topo-vector, but not entirely because it includes a raster component (the hillshade) and so it behaves as raster from a scale perspective.

The problem is that the integration between deck.gl and ArcGIS is still based on this view class: https://deck.gl/docs/api-reference/core/map-view We really shouldn't; I hope to switch to a different integration strategy soon but I haven't been able so far, sorry :-(

Anyway, there is nothing wrong with `map-view`; but it is modelled after MapBox GL. For the ArcGIS/deck.gl 2D integration, we have a reliable way to determine the zoom level from the scale in the WebMercator case; we can do this both for vector tile basemaps (where the behaviors of the ArcGIS JS API/MapBox/Deck.GL all match) and raster basemaps, where we are off by a factor of 2 but it can be detected.

We need to do the same for 3D. I am in contact with my colleagues that work on 3D, hopefully we can fix it soon. In the meantime, I have been experimenting in a local branch of mine. Basically, if you inject a scale factor (1 or 2, depending on the basemap) from the app, any basemap can be made to work; see [this](https://github.com/visgl/deck.gl/compare/master...damix911:damix911/arcgis3dzoom#diff-4875135ae4a8bba91ec85157d5e5b4b2788bdb939440c8c99bd9dff5a798c560R19-R20) and [this](https://github.com/visgl/deck.gl/compare/master...damix911:damix911/arcgis3dzoom#diff-4875135ae4a8bba91ec85157d5e5b4b2788bdb939440c8c99bd9dff5a798c560R56).

If it helps in the meantime, you can check out my branch, and `yarn bootstrap`, `cd examples/get-started/pure-js/arcgis-3d`, `yarn`, `yarn start-local`.

We need to make this automatic.

![modern-antique](https://user-images.githubusercontent.com/12469086/165180765-2d975e43-7c21-43cc-ad96-04fc5048f1f1.png)
![streets-vector](https://user-images.githubusercontent.com/12469086/165180770-6cdaed1c-f0ea-4c34-885e-4c8b3792962d.png)
![streets](https://user-images.githubusercontent.com/12469086/165180772-942ea7f4-3520-4939-8d2a-e092b78a9746.png)
![nova](https://user-images.githubusercontent.com/12469086/165180773-8148c146-87b6-4f36-ad9e-4c2e0ce40255.png)
![topo](https://user-images.githubusercontent.com/12469086/165180775-9007159a-4210-438b-bbdb-f8decb97aa6f.png)
![topo-vector](https://user-images.githubusercontent.com/12469086/165180777-26189bef-20d4-41ab-b0cb-57e545d5db06.png)
![satellite](https://user-images.githubusercontent.com/12469086/165180781-cb2024ed-5c55-4669-abf7-9d9dac20b13b.png)
![dark-gray](https://user-images.githubusercontent.com/12469086/165180783-9fd8e9bc-5f25-48ab-bedd-8e8955f8eb93.png)
![dark-gray-vector](https://user-images.githubusercontent.com/12469086/165180786-ad8ef6e6-50b6-430d-a2bb-25f0435296d5.png)

**Comment by Brave-Jun** [comment]
HI, is there any progress on this issue, I test the problem still exists. Thank you!

**Comment by mnsr002** [comment]
Hi team, any update on this issue?

**Comment by GemT27** [comment]
Hi team, any update on this issue?
